src/shared.ts

```diff
@@ -66,3 +58,11 @@ export function registerProxy(proxy: object, value: any) {
 }
 
 export const unwrap = (replicaOrAny: any): any => proxyToValueMap.get(replicaOrAny) || replicaOrAny;
+
+export function getOwnKeysAndMagicProperty(target: object, magicPropertyKey: PropertyKey | undefined): PropertyKey[] {
+    const keys = isUndefined(magicPropertyKey) || hasOwnProperty.call(target, magicPropertyKey) ? [] : [magicPropertyKey];
```

Comment:
this method follows @jdalton suggestion to avoid the creation of an extra array when doing concat. it also adds the magic key when present.